app/controllers/application_controller.rb

```diff
@@ -6,16 +6,63 @@ def log_in(user)
   end
 
   def log_out
+    forget(current_user)
     session.delete(:user_id)
     @current_user=nil
   end
 
+  def forget(user)
+    user.forget
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end 
+
+
+  helper_method :current_user
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user && user.authenticated?(:remember, cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end 
```

Comment:
Ahh yes, I see. No, there is no need to also attempt to store `user_id` into the session if it's already in the cookie (which is the point of the conditionals anyway.